### ISSUE TITLE
[Python] Implement new Concept Exercise: generators

### ISSUE THREAD

**Issue by BethanyG** [exercism/v3#1378]
This issue describes how to implement the `generators` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the syntax and use of `generators` in Python. 

## Learning objectives

-  Understand what generators are and how/when to use them
-  Understand how generators relate to `loops` and `iterators`
-  Understand how to use the `yield` keyword
-  Understand the `__next__()` method
-  Create a generator

## Out of scope

- Memory and performance characteristics and optimizations
- `throw(type, value=None, traceback=None)`
- `close()`
- `generator expressions`
- `generators` used as coroutines

## Concepts

-  `generators`
-  `yield`
-   `__next__()`
-  `iterators`

## Prerequisites

- [for-loops](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/for_loops.md)
- [builtin-types](https://github.com/exercism/v3/tree/master/languages/python/reference/concepts/builtin_types)
- [data-types](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/data_structures.md)
- [iterables](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iterables.md)
- [iterators](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iterators.md)
- [iteration](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iteration.md)

## Resources to refer to

-  [Generators (Python official docs)](https://docs.python.org/3/howto/functional.html#generators)
- [generator (Python official docs glossary)](https://docs.python.org/3/glossary.html#term-generator)
- [The yield statement (Python official docs)](https://docs.python.org/3/reference/simple_stmts.html#the-yield-statement)
- [Yield expressions (Python official docs)](https://docs.python.org/3/reference/expressions.html#yieldexpr)
- [Iterators(Python official docs)](https://docs.python.org/3/howto/functional.html?#iterators)
- [Generator-iterator methods (Python official docs)](https://docs.python.org/3/reference/expressions.html#generator-iterator-methods)
- [How to Use Generators and yield in Python (Real Python)](https://realpython.com/introduction-to-python-generators/)

### Hints

- `Generators` section of the Python Docs Functional How to tutorial:  [Generators](https://docs.python.org/3/howto/functional.html#generators)

### After

-  Additional `Generator-iterator methods`, such as `generator.send()` and `generator.throw()`
- `generator expressions`
-  Asynchronous generator functions
- `generators` used as coroutines

## Representer

No changes required.

## Analyzer

No changes rquired.

## Implementing

- Tests should be written using unittest.TestCase, and the test file named list_comprehensions_test.py.
- [How to Implement a Concept Exercise in Python](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md)
- [make-concept-exercise Utility](https://github.com/exercism/v3/tree/master/languages/python/bin)

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
